src/shutdown.cc

```diff
@@ -499,7 +538,19 @@ void do_system_shutdown(shutdown_type_t shutdown_type)
 #ifdef __NetBSD__
     reboot(reboot_type, NULL);
 #else
-    reboot(reboot_type);
+    if (reboot(reboot_type)) {
+        // we're in trouble now
+        sub_buf.append("reboot: ");
+        sub_buf.append(strerror(errno));
+        if (shutdown_type == shutdown_type_t::KEXEC) {
+            sub_buf.append(
+                    "\nIt is possible that no suitable kernel image was loaded before"
+                    "\nreboot with kexec was attempted.\n"
+            );
+        }
+        while (true)
+            loop.run();
```

Comment:
Code style:

    If curly braces are omitted (for a control statement) then the entire statement should be a
    single line. Otherwise the braces are mandatory.

Either use braces or combine the `while` into a single line, please.